src/Mod/Path/PathScripts/PathVcarve.py

```diff
@@ -0,0 +1,280 @@
+                obj.SafeHeight.Value)))
+            c = Path.Command("G1 X{} Y{} Z{} F{}".format(p.x, p.y, p.z,
+                obj.ToolController.HorizFeed.Value))
+            path.append(c)
+            for e in w.Edges:
+                path.extend(PathGeom.cmdsForEdge(e,
+                    hSpeed=obj.ToolController.HorizFeed.Value))
+
+            return path
+
+        pathlist = []
+        pathlist.append(Path.Command("(starting)"))
+        for f in Faces:
+            vd = Path.Voronoi()
+            insert_many_wires(vd, f.Wires)
+
+            vd.construct()
+
+            for e in vd.Edges:
+                e.Color = 0 if e.isPrimary() else 5
+            vd.colorExterior(1)
+            vd.colorExterior(4, lambda v: not f.isInside(v.toGeom(),
+                obj.Tolerance, True))
+            vd.colorColinear(3, obj.Threshold)
+            vd.colorTwins(2)
+
+            edgelist = getEdges(vd)
+
+            for wire in getWires(edgelist):
+                pathlist.extend(cutWire(wire))
+
+        self.commandlist = pathlist
+
+    def opExecute(self, obj):
+        '''opExecute(obj) ... process engraving operation'''
+        PathLog.track()
+
+        if obj.ToolController.Tool.CuttingEdgeAngle >= 180.0:
+            FreeCAD.Console.PrintError(
+                translate("Path_Vcarve",
+                    "Engraver Cutting Edge Angle must be < 180 degrees.") + "\n")
+            return
+        try:
+            if obj.Base:
+                PathLog.track()
+                for base in obj.Base:
+                    faces = []
+                    for sub in base[1]:
+                        shape = getattr(base[0].Shape, sub)
+                        if isinstance(shape, Part.Face):
+                            faces.append(shape)
+
+                modelshape = Part.makeCompound(faces)
+
+            elif len(self.model) == 1 and self.model[0].isDerivedFrom('Sketcher::SketchObject') or \
+                    self.model[0].isDerivedFrom('Part::Part2DObject'):
+                PathLog.track()
+
+                modelshape = self.model[0].Shape
+            self.buildPathMedial(obj, modelshape.Faces)
+
+        except Exception as e:
+            PathLog.error(e)
+            traceback.print_exc()
+            PathLog.error(translate('PathVcarve',
+                'The Job Base Object has no engraveable element.\
+                        Engraving operation will produce no output.'))
+
+    def opUpdateDepths(self, obj, ignoreErrors=False):
+        '''updateDepths(obj) ... engraving is always done at \
+                the top most z-value'''
+        job = PathUtils.findParentJob(obj)
+        self.opSetDefaultValues(obj, job)
+
+
+def SetupProperties():
+    return ["Discretize"]
+
+
+def Create(name, obj=None):
+    '''Create(name) ... Creates and returns a Vcarve operation.'''
+    if obj is None:
+        obj = FreeCAD.ActiveDocument.addObject("Path::FeaturePython", name)
+    proxy = ObjectVcarve(obj, name)
```

Comment:
There is no need for the return value of a Proxy class.

Reply:
fixed